### PR TITLE
Added support for access control on templates

### LINKED ISSUE #2370: Permissions Branch - Users have access to custom templates that were previously created in private channels
## Steps to reproduce the behavior

After updating, custom templates that were created in workspaces tied to private channels become accessible to any user that's part of that same team to which that private channel/workspace belonged to. 

Is this expected?

## Edition and Platform

- permissions-feature-branch

### LINKED ISSUE #2377: Permissions branch: Share dialog role selector is cut off
## Steps to reproduce the behavior

1. Open a board
2. Select "Show Share Dialog"
3. Select the role for your user
4. The menu is cut off

## Expected behavior

The menu should be visible

## Screenshots

Screen Shot 2022-02-22 at 3 52 15 PM

## Edition and Platform

- Edition: Mattermost plugin
- Version: permissions-feature-branch
- Browser and OS: Chrome on Mac

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
1. Added a separate API for fetching system default templates.
2. Updated `handleGetTemplates` API to include following boards, scoped to the current team-
    * all public templates,
    * all private templates where the user is added as a member.
3. Updated share board modal to hide sharing URL sections for templates.

#### Ticket Link
Fixes #2370
Fixes #2377

### CODE REVIEW

File: server/api/api.go [hunk 1]
```diff
@@ -2208,6 +2209,62 @@ func (a *API) handleGetTemplates(w http.ResponseWriter, r *http.Request) {
 	auditRec.Success()
 }
 
+func (a *API) handleGetDefaultTemplates(w http.ResponseWriter, r *http.Request) {
```
Comment (wiggin77) [inline server/api/api.go:2212]: Not sure I understand the reasoning to make this a separate API versus fetching templates from team 0.

Comment (harshilsharma63) [reply server/api/api.go]: Yeah, this is not needed. It was needed in the middle of the implementation but I changed something and now it's not needed anymore.

Removed the API.

**Review by sbishel — COMMENTED** [review]
I think the question Doug was asking (if not, I am :) ) is why we need `GetDefaultTemplates` at all. Shouldn't calling `GetTemplateBoards({teamID: '0', user: 'abc'}) provide the same results? If not , why. I think keeping templates working as similar to boards as possible is important.

File: server/services/store/sqlstore/templates.go [hunk 2]
```diff
@@ -71,3 +71,41 @@ func (s *SQLStore) getTemplateBoards(db sq.BaseRunner, teamID string) ([]*model.
 
 	return s.boardsFromRows(rows)
 }
+
+// getDefaultTemplateBoards fetches all template blocks .
```
[the remark this answers is not in the document]
Comment (harshilsharma63) [reply server/services/store/sqlstore/templates.go]: Done.

**Comment by harshilsharma63** [comment]
> I think the question Doug was asking (if not, I am :) ) is why we need `GetDefaultTemplates` at all. Shouldn't calling `GetTemplateBoards({teamID: '0', user: 'abc'}) provide the same results? If not , why. I think keeping templates working as similar to boards as possible is important.

Yeah, turns out `GetDefaultTemplates` is not needed, neither in the API nor anywhere else. Removed it now.

File: server/app/onboarding.go [hunk 3]
```diff
@@ -46,14 +46,14 @@ func (a *App) PrepareOnboardingTour(userID string, teamID string) (string, strin
 }
 
 func (a *App) getOnboardingBoardID() (string, error) {
-	boards, err := a.store.GetTemplateBoards(globalTeamID)
+	boards, err := a.store.GetTemplateBoards("0", "")
```
Comment (sbishel) [inline server/app/onboarding.go:49]: Use globalTeamID instead of "0". Here and elsewhere.

**Review by sbishel — CHANGES_REQUESTED** [review]

File: server/app/onboarding.go (hunk 3 above)
Comment (harshilsharma63) [reply server/app/onboarding.go]: Done.

**Review by sbishel — APPROVED** [review]

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

**Pull request merged**